mantidproject/mantid
# Workbench: Project Save - Workspaces Only

**Description of work.**
This is an initial implementation of Project Saving, it only currently handles workspaces. I have changed the File menu around a bit to make it fit better with the new additions and changed the names of the calls to open and save scripts.

I understand that there are some pieces of code that don't necessarily do anything right now however, it's intended to be integrated further with different mechanics for the full project saving later down the line.

**To test:**
- Launch workbench
- Load a workspace
- File->Save Project and choose a place to save your project folder
- File->Save Project as and choose a different place to save your project folder
- Close workbench and relaunch workbench
- File->Open Project and select the first saved project it should load those workspaces back successfully and no errors should show in the logs
- Close and relaunch workbench
- File->Open project and select the second saved project and it also should load those exact same workspaces back successfully and no errors should show in the logs


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
